### PR TITLE
Various bug fixes for plant selection, plant placement and select box during placement mode

### PR DESCRIPTION
**Pull request by tschawax**
## Basics

- [X] I added a line to [changelog.md](/doc/changelog.md)
- [X] The PR is rebased with current master.
- [X] Details of what you changed are in commit messages.
- [X] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

**Comment by markus2330** [comment]
jenkins build please

File: doc/changelog.md [hunk 1]
```diff
@@ -61,6 +61,9 @@ Syntax: `- short text describing the change _(Your Name)_`
 - _()_
 - Add Christoph Nemeth as Teammember _(tschawax)_
 - Fix middle mouse button to only move the stage or select individual plants _(tschawax)_
+- Fix selections being kept after choosing a plant from the search list _(tschawax)_
```
Comment (markus2330) [inline doc/changelog.md:64]: Better to add issue numbers here (and not as comment in the PR).

File: frontend/src/features/map_planning/components/BaseStage.tsx [hunk 2]
```diff
@@ -172,6 +173,8 @@ export const BaseStage = ({
       return;
     }
 
+    if (isPlacementModeActive()) return;
```
Comment (markus2330) [inline frontend/src/features/map_planning/components/BaseStage.tsx:176]: Maybe write a comment what this early exit does.

File: frontend/src/features/map_planning/layers/plant/components/PlantingElement.tsx [hunk 3]
```diff
@@ -23,6 +25,17 @@ const placeTooltip = (plant: PlantsSummaryDto | undefined) => {
...
 };
 
+/**
+ * UI Component representing a single plant on the map.
+ *
+ * A single plant consists of a _Konva Group_ which itself is composed of:
+ * - A _Konva Circle_ filled with a selection-dependent colour
+ * - A _Konva Image_ depicturing the plant or showing a fallback image
+ *
+ * @param planting - A plant's details used for rendering it.
```
Comment (markus2330) [inline frontend/src/features/map_planning/layers/plant/components/PlantingElement.tsx:35]: Is it plural or singular? What is it?

```suggestion
 * @param planting - A plant's detail used for rendering the plant.
```

```suggestion
 * @param planting - Plant's details used for rendering the plant.
```

**Review by markus2330 — APPROVED** [review]
Great job, works much better now. @yvonnemarkl is also very happy.

**Review by badnames — APPROVED** [review]

**Pull request merged**